File: app/src/main/java/ua/dp/rename/dniprostreets/di/module/AppModule.java
package ua.dp.rename.dniprostreets.di.module;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ua.dp.rename.dniprostreets.App;

@Module
public class AppModule {

    private final App application;

    public AppModule(App application) {
        this.application = application;
    }

    @Provides @Singleton
    Context provideApplicationContext() {
        return application;
    }
}

File: app/src/main/java/ua/dp/rename/dniprostreets/di/module/AppModule.kt
package ua.dp.rename.dniprostreets.di.module

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import ua.dp.rename.dniprostreets.di.AppContext
import javax.inject.Singleton

@Module
class AppModule(private val app: Application) {

   @Singleton
   @Provides
   @AppContext
   fun provideAppContext(): Context = app
}
